src/receivers/otlp_http.rs

```diff
@@ -592,13 +586,22 @@ fn response_4xx_with_body(
         .unwrap())
 }
 
-fn compute_ok_resp<T: prost::Message + Default>() -> Result<Bytes, EncodeError> {
+fn compute_ok_resp<T: prost::Message + Serialize + Default>(
+    as_json: bool,
+) -> Result<Bytes, EncodeError> {
     // The default response is actually empty, so this results in an empty response
     let resp = T::default();
-    let mut buf = Vec::with_capacity(resp.encoded_len());
-    resp.encode(&mut buf)?;
 
-    Ok(buf.into())
+    let ret_buf = match as_json {
+        true => serde_json::to_vec(&resp).unwrap(),
+        false => {
+            let mut buf = Vec::with_capacity(resp.encoded_len());
+            resp.encode(&mut buf)?;
+            buf
+        }
+    };
+
```

Comment:
These messages are zero bytes for PB and minimal JSON. We could pre-materialize like we were before, but unlikely there's a big benefit.

Reply:
OK, well we can always pprof it on Monday vs and old build. The JSON one probably has a little more of a hit but ideally no one would run that at scale so agreed.